dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/internal/BaseDimensionDataCloudControlMockTest.java

```diff
@@ -68,10 +76,42 @@ public void start() throws IOException {
             .endpoint(url("")).modules(modules).overrides(overrides()).build();
       json = ctx.utils().injector().getInstance(Json.class);
       api = ctx.getApi();
+      applyAdditionalServerConfig();
+      assertedRequestCount = 0;
+   }
+
+   /**
+    * Applies any additional configuration required by test classes to the mock web server.
+    */
+   protected void applyAdditionalServerConfig() {
+   }
+
+   @Override
+   public void run(IHookCallBack callBack, ITestResult testResult) {
+      callBack.runTestMethod(testResult);
+      ensureAllRequestsWereAsserted();
+   }
+
+   private void ensureAllRequestsWereAsserted() {
+      int unAssertedRequestCount = server.getRequestCount() - assertedRequestCount;
+      if (unAssertedRequestCount > 0) {
+         StringBuilder messageBuilder = new StringBuilder(
+               String.format("There were %s un-asserted requests: ", unAssertedRequestCount));
+         while (unAssertedRequestCount > 0) {
+            try {
+               messageBuilder.append('\n').append(server.takeRequest().toString());
+            } catch (InterruptedException e) {
+               messageBuilder.append('\n').append("Error obtaining request from mock web server: ")
+                     .append(e.toString());
+            }
+            unAssertedRequestCount--;
+         }
+         fail(messageBuilder.toString());
+      }
    }
 
    @AfterMethod(alwaysRun = true)
-   public void stop() throws IOException {
+   public void stop(ITestContext testContext) throws Exception {
```

Comment:
Nuts - no it's not - put it in whilst trying something else and forgot it was there!